lib/src/view/game/game_screen.dart

```diff
@@ -68,12 +69,69 @@ class _GameScreenState extends ConsumerState<GameScreen> {
   final _whiteClockKey = GlobalKey(debugLabel: 'whiteClockOnGameScreen');
   final _blackClockKey = GlobalKey(debugLabel: 'blackClockOnGameScreen');
   final _boardKey = GlobalKey(debugLabel: 'boardOnGameScreen');
+  AppLifecycleListener? _appLifecycleListener;
+
+  @override
+  void initState() {
+    super.initState();
+    // Cancel pending seek or challenges when app goes to background to prevent games being created
+    // while the user is away from the app.
+    _appLifecycleListener = AppLifecycleListener(onPause: _cancelSeekOrChallenge);
+  }
+
+  @override
+  void dispose() {
+    _appLifecycleListener?.dispose();
+    super.dispose();
+  }
+
+  Future<void> _cancelSeekOrChallenge() async {
+    if (!mounted) return;
+    final loader = ref.read(gameScreenLoaderProvider(widget.source));
+    // Only cancel if we are still seeking or waiting for challenge to be accepted
+    if (loader is! AsyncLoading<GameScreenState>) {
+      return;
+    }
+    switch (widget.source) {
+      case LobbySource():
+        ref.read(gameScreenLoaderProvider(widget.source).notifier).cancelSeek();
+        await ref.read(createGameServiceProvider).cancelSeek();
+      case UserChallengeSource():
+        ref.read(gameScreenLoaderProvider(widget.source).notifier).cancelChallenge();
+        await ref.read(createGameServiceProvider).cancelChallenge();
+      case ExistingGameSource():
+        break;
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
     final boardPreferences = ref.watch(boardPreferencesProvider);
 
     switch (ref.watch(gameScreenLoaderProvider(widget.source))) {
+      case AsyncData(value: SeekCancelledState()):
+        return Scaffold(
+          resizeToAvoidBottomInset: false,
+          appBar: AppBar(
+            leading: const SocketPingRatingIcon(),
```

Comment:
Instead of showing the back button in the bottom bar, remove this widget to show it in the top bar as it should be.